src/multivector.jl

```diff
@@ -13,15 +13,15 @@ Coefficients are stored using a sparse coding, and only the coefficients of the
 """
 struct MultiVector{CA,T,BI,K}
     c::NTuple{K,T}
-    function MultiVector(
-        CA::Type{<:CliffordAlgebra},
-        BI::NTuple{K,Integer},
-        c::NTuple{K,T},
-    ) where {K,T<:Real}
-        @assert length(BI) > 0 
-        #@assert issorted(BI) && allunique(BI)
-        new{CA,T,convert(NTuple{K,Int}, BI),K}(c)
-    end
+end
+function MultiVector(
+    CA::Type{<:CliffordAlgebra},
+    BI::NTuple{K,Integer},
+    c::NTuple{K,T},
+) where {K,T<:Real}
+    @assert length(BI) > 0 
+    #@assert issorted(BI) && allunique(BI)
+    MultiVector{CA,T,convert(NTuple{K,Int}, BI),K}(c)
 end
```

Comment:
The old inner constructor relies on const prop for inference. However, const prop is not as robust as type stable code. Especially in older julia versions like 1.6 I found it impossible to fix type inference with the old inner constructor.